shuup/xtheme/rendering.py

```diff
@@ -117,22 +118,36 @@ def render(self):
         :return: Rendered markup.
         :rtype: markupsafe.Markup
         """
+        language = get_language()
+
         if not may_inject(self.context):
             return ""
 
         full_content = ""
         for layout in self.layouts:
-            wrapper_start = "<div%s>" % get_html_attrs(self._get_wrapper_attrs(layout))
-            buffer = []
-            write = buffer.append
-            self._render_layout(write, layout)
-            content = "".join(buffer)
-            full_content += (
-                "%(wrapper_start)s%(content)s%(wrapper_end)s" % {
-                    "wrapper_start": wrapper_start,
-                    "content": content,
-                    "wrapper_end": "</div>",
-                })
+            cache_key = "shuup_xtheme_placeholder_%s_%s_%s_%s" % (
+                self.view_config.saved_view_config.pk,
+                language,
+                self.placeholder_name,
+                get_layout_data_key(self.placeholder_name, layout, self.context)
+            )
+            layout_content = cache.get(cache_key)
```

Comment:
have you really tested this? what if we change some property of the plugin, will it be updated? shouldn't we bump the cache when we update/create a SavedViewConfig?

Reply:
yes that why `saved_view_config.pk` is in cache key. It bumps whenever you do any changes with the editor for the view.